<a href="https://colab.research.google.com/github/joshlensing/Coding-Dojo-Project-2/blob/main/Project_Proposal.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project Proposal

- Name: Joshua Lensing
- Coding Dojo Data Science Course

## Proposal Number 1

For my first project proposal, I want to use baseball stats from baseball teams from 1962-2012 in order to predict whether teams will make the playoffs or not. I am a very big fan of baseball and have been since I was 8 years old. Baseball data and analytics are becoming a very big component of how professional baseball teams operate and I would love to someday be working in those departments for a major league club. A lot of this dataset comes from the idea of "Moneyball," where Billy Beane, the GM of the Oakland Athletics, found statisitics relating to individual player performance that scouts and other front offices were not focused on at the time. He hoped to use those background statistics and develop a formula for success that didn't require spending a large amount of money to build a roster of players. Instead, he hoped to find players that scored well in these statistical categories that other teams would often overlook and pay them cheaply to play for the Athletics.

1. Source of the dataset: https://www.kaggle.com/datasets/wduckett/moneyball-mlb-stats-19622012


In [2]:
import pandas as pd

In [5]:
df = pd.read_csv('/content/baseball.csv')
df.head()

Team League  Year   RS   RA   W    OBP    SLG     BA  Playoffs  RankSeason  \
0  ARI     NL  2012  734  688  81  0.328  0.418  0.259         0         NaN   
1  ATL     NL  2012  700  600  94  0.320  0.389  0.247         1         4.0   
2  BAL     AL  2012  712  705  93  0.311  0.417  0.247         1         5.0   
3  BOS     AL  2012  734  806  69  0.315  0.415  0.260         0         NaN   
4  CHC     NL  2012  613  759  61  0.302  0.378  0.240         0         NaN   

   RankPlayoffs    G   OOBP   OSLG  
0           NaN  162  0.317  0.415  
1           5.0  162  0.306  0.378  
2           4.0  162  0.315  0.403  
3           NaN  162  0.331  0.428  
4           NaN  162  0.335  0.424

In [6]:
df.shape

(1232, 15)

2. The data shows the stats from teams of both the American and National Leagues and from each year from 1962-2012. The stats strictly show hitting stats, team wins, and teams runs scored and allowed. It also has a column for if the team made the playoffs.

3. **What is the target?** The target of the data is the 'Playoffs' column. This determines whether or not the team made the playoffs or not.

4. **What does one row represent?** One row of the dataset represents a team.

5. **Is this a classification or regression problem?** The target is a classification problem, as I am looking for 1 of 2 outcomes.

6. **How many features does the data have?** It has 14 features, a couple of which may end up being dropped ('RankSeason' and 'RankPlayoff').

7. **How many rows are in the dataset?** There are 1,232 rows.

8. **What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?** The statistical data all appears to be intact for each row, except for the final 2 columns, which are unclear as to what they are as they already have a similar column in the dataset. I don't anticipate too many hurdles with this dataset.

##Proposal Number 2

My second project proposal has a couple different meanings to me. I went to school for my neuroscince degree and my grandfather died from complications due to Alzheimer's. My dad also currently has Parkinson's disease, so neurological disorders hit very close to home with me. I would have chosen a Parkinson's disease dataset as my first proposal, but I could not find a dataset large enough to use.

I currently work in for a biotech company as a small rodent surgeon and I feel that finding a job in the biomedical community may also be a great fit. Combining both of those aspects of my life together makes for a very interesting project.

1. Source: https://www.kaggle.com/datasets/brsdincer/alzheimer-features

In [11]:
df = pd.read_csv('/content/alzheimer.csv')
df.head()

Group M/F  Age  EDUC  SES  MMSE  CDR  eTIV   nWBV    ASF
0  Nondemented   M   87    14  2.0  27.0  0.0  1987  0.696  0.883
1  Nondemented   M   88    14  2.0  30.0  0.0  2004  0.681  0.876
2     Demented   M   75    12  NaN  23.0  0.5  1678  0.736  1.046
3     Demented   M   76    12  NaN  28.0  0.5  1738  0.713  1.010
4     Demented   M   80    12  NaN  22.0  0.5  1698  0.701  1.034

In [12]:
df.shape

(373, 10)

2. The data contains measurements of brain volume, a mental state examination, age, socioeconomic status, and years of education. All of these are compiled together with a determination if they have Alzheimer's disease or not.

3. **What is the target?** The target is the 'Group' column, which determines if the person has been diagnosed with Alzheimer's disease or not. 'Demented' would be the positive class and 'Nondemented' would be the negative class.

4. **What does one row represent?** One row represents a person.

5. **Is this a classification problem or regression?** This is a classification problem.

6. **How many features does the data have?** The data has 8 features.

7. **How many rows are in the dataset?** There are 373 rows.

8. **What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?** The dataset may not be big enough with 8 features and 373 rows. It may not allow the models to score very well in their predictions. There is also some missing data as well.